sdk/core/azure-core/src/http/curl/curl_connection_private.hpp

```diff
@@ -201,6 +218,23 @@ namespace Azure { namespace Core {
        */
       bool IsExpired() override
       {
+        // if we have keep alive options and we haven reached the max requests declare expired
+        if (m_keepAliveOptions.HasValue() && m_keepAliveOptions.Value().MaxRequests > 0
+            && m_keepAliveOptions.Value().MaxRequests > m_usedCount)
+        {
+          return true;
+        }
+
+        // if we have keep alive options and we have a connection timeout and the connection time
+        // frame has passed declare expired
+        if (m_keepAliveOptions.HasValue()
```

Comment:
I am pretty sure that this code says that as long as we're less than MaxRequests, we'll ignore the connection timeout. Is that your intent?

Reply:
done